user_service/internal/repository/user_repository.go

```diff
@@ -61,6 +61,16 @@ func (repo *UserRepository) GetUserByEmail(userEmail string) (*models.User, erro
 	return user, nil
 }
 
+func (repo *UserRepository) ValidateUserExistance(userEmail string) (*models.User, error) {
+	user := &models.User{}
+	err := repo.dbClient.Debug().Where("email = ?", userEmail).First(user).Error
+	if err != nil {
```

Comment:
In caz ca din oarecare conditii sunt alte erori in baza de date, care nu e legata de record is not found, atunci mai departe la serviciu se duce eroarea data, unde se va trimite iar request la repo sa creeze un user. La service ar trebui sa se compare eroarea venita cu record is not found si numai in cazul dat sa mearga mai departe la crearea utilizatorului